content-security-policy/script-src/script-src-strict_dynamic_double_policy_different_nonce.html

```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>A separate policy with more nonces works correctly with `strict-dynamic` in the script-src directive.</title>
+    <script src='/resources/testharness.js' nonce='dummy'></script>
+    <script src='/resources/testharnessreport.js' nonce='dummy'></script>
+    <script src='../support/checkSecurityPolicyViolations.sub.js' nonce='dummy'></script>
+    <!-- CSP served:
+    1) Content-Security-Policy: script-src 'strict-dynamic' 'nonce-dummy'
+    2) Content-Security-Policy: script-src 'nonce-dummy' 'nonce-dummy2'
+    -->
+</head>
+
+<body>
+    <h1>A separate policy with more nonces works correctly with `strict-dynamic` in the script-src directive.</h1>
+    <div id='log'></div>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            var e = document.createElement('script');
+            e.src = 'simpleSourcedScript.js';
+            e.onload = t.unreached_func('OnLoad should not be triggered.');
+            document.body.appendChild(e);
+            t.done();
+        }, 'Unnonced script injected via `appendChild` is not allowed with `strict-dynamic` + a nonce-only double policy.');
+    </script>
+
+    <script nonce='dummy'>
+        var simpleSourcedScriptRan = false;
+        async_test(function(t) {
+            var e = document.createElement('script');
+            e.setAttribute('nonce', 'dummy2');
+            e.src = 'simpleSourcedScript.js';
+            e.onload = t.step_func_done(function() {
+                assert_true(simpleSourcedScriptRan);
```

Comment:
Hrm. You have two tests looking at the same global state. Would it be possible to have the script alter its environment in some non-shared way? For instance: `document.currentScript.setAttribute('executed', '1');` followed by `assert_equals(e.getAttribute('executed'), '1');`.